### ISSUE TITLE
Classify and optionally drop some redundant template-related nodes and edges [phab:D790]

### ISSUE THREAD

**Issue by kythebot** [google/kythe#1079]
_Created by **zarko** at 2016-03-23 22:53:56_: 

This CL classifies childof edges that relate anchors to template
instantiations as "childof/context" edges. It also provides an
experimental indexer flag that drops anchors and edges that are
template instantiation independent. For example, in the program:

```
  template<typename T> class C { int i; };
  template class C<short>;
```

the anchor covering `i` will be emitted twice, once in `class C`
and once in `C<short>` with a `childof/context` edge to `C<short>`.
Similarly, the anchor covering `int` and its `ref` edge will be emitted
twice unless the new option is asserted, in which case only those related
to the primary template will be emitted.

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-03-24 16:56:02_:

minor, but I find operators using `std::tie()` more readable, e.g.

    std::tie(Hash, PhysicalRange, EdgeKind, EdgeTarget) ==
    std::tie(other.Hash, other.PhysicalRange, other.EdgeKind, other.EdgeTarget);

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-03-24 16:56:02_:

at 8 positional arguments it's probably worth naming them and doing better argument parsing;

TEST_FILE="$1"

while shift; [ $# -ne 0]; do
  case "$1" in
    --expected)
      RESULTS_EXPECTED="$2"
      shift
      ;;
    # etc.
  esac
done

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-24 16:56:02_:

a couple of minor comments.

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-24 18:16:50_:

  - Address comments.

**Comment by kythe** [comment]
_**shahms** accepted this revision at 2016-03-24 20:14:47_

**Comment by kythe** [comment]
_**zarko** closed via commit e695fbbdb2131e1bba8bbe8560975b305e4ed951 at 2016-03-24 20:32:43_
